apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -0,0 +1,252 @@
+      __csrf_token: token,
+      course_sharing_name: name,
+    }).toString(),
+  });
+}
+
+async function accessSharedQuestionAssessment() {
+  const assessmentsUrl = `${baseUrl}/course_instance/${exampleCourseId}/instructor/instance_admin/assessments`;
+  const assessmentsPage = await helperClient.fetchCheerio(assessmentsUrl);
+  const sharedQuestionAssessmentUrl =
+    siteUrl +
+    assessmentsPage
+      .$(`a:contains("Example of Importing Questions From Another Course")`)
+      .attr('href');
+  const res = await helperClient.fetchCheerio(sharedQuestionAssessmentUrl);
+  assert.equal(res.ok, true);
+  return res;
+}
+
+describe('Question Sharing', function () {
+  this.timeout(80000);
+
+  describe('Create a sharing set and add a question to it', () => {
+    let exampleCourseSharingId;
+
+    step('set up testing server', helperServer.before(TEST_COURSE_PATH));
+    after('shut down testing server', helperServer.after);
+
+    step('ensure course has question sharing enabled', async () => {
+      await sqldb.queryAsync(sql.enable_question_sharing, {});
+    });
+
+    step('Sync coures with sharing enabled', (callback) => {
+      syncFromDisk.syncOrCreateDiskToSql(EXAMPLE_COURSE_PATH, logger, function (err, result) {
+        if (ERR(err, callback)) return;
+        // TODO: technically this would have an error because there is no permissions on the
+        // shared question, but we are configured to ignore sharing errors locally. Is this the right thing to do?
+        if (result?.hadJsonErrorsOrWarnings) {
+          return callback(
+            new Error(`Errors or warnings found during sync of ${EXAMPLE_COURSE_PATH}`)
+          );
+        }
+        callback(null);
+      });
+    });
+
+    step(
+      'Fail to access shared question, because permission has not yet been granted',
+      async () => {
+        const res = await accessSharedQuestionAssessment();
+        // TODO: Now that we add a dummy question to the DB,
+        // then the name of it will show up, but it should fail to load when you access the link
+        // this should be updated to actually attempt to go to the link, then hit access denied or something
+        assert(!res.text().includes('addNumbers'));
+        // const sharedQuestionUrl = siteUrl + res.$(`a:contains("Add two numbers")`).attr('href');
+        // let addNumbersPage = await helperClient.fetchCheerio(sharedQuestionUrl);
+        // assert(!addNumbersPage.ok);
+      }
+    );
+
+    step('Fail if trying to set an invalid sharing name', async () => {
+      let res = await setSharingName(testCourseId, 'invalid@sharingname');
+      assert(res.status === 400);
+
+      res = await setSharingName(testCourseId, 'invalid / sharingname');
+      assert(res.status === 400);
+    });
+
+    step('Set test course sharing name', async () => {
+      await setSharingName(testCourseId, testCourseSharingName);
+      const sharingPage = await (await fetch(sharingPageUrl(testCourseId))).text();
```

Comment:
It's important to check `res.status` or `res.ok` in tests to ensure that we didn't hit an error page that happened to include the desired text.

Reply:
done